server/channels/app/post.go

```diff
@@ -977,6 +1018,11 @@ func (a *App) PatchPost(rctx request.CTX, postID string, patch *model.PostPatch,
 		return nil, err
 	}
 
+	// only allow to update the pinned status of burn-on-read posts if the status is different
+	if post.Type == model.PostTypeBurnOnRead && model.SafeDereference(patch.IsPinned) == post.IsPinned {
```

Comment:
Same as the UpdatePost function, I don't fully understand this. We're allowing to patch if BoR's pinned status didn't change?